### DISCUSSION TITLE
One-to-many example

### DISCUSSION THREAD

**Discussion by LarsKrogJensen** [operator-framework/java-operator-sdk#1522] [General]
Hi,

I am looking at solving a custom controller that needs to map one CR to several (1..n) secondary resources (ServiceEntry), cannot seem to find any tests, samples or docs around this scenario.

**Comment by csviri** [comment]
@LarsKrogJensen using the low level API or Dependent Resources ?

**Reply by LarsKrogJensen** [reply]
Preferably Dependent Resource

**Reply by csviri** [reply]
That is not released yet, see: https://github.com/java-operator-sdk/java-operator-sdk/issues/1182
Will come in 4.1. See related integration tests in the PR. Note that this API will change a little bit, if you are interested:  https://github.com/java-operator-sdk/java-operator-sdk/pull/1521 
But probably more improvements will come.


With low level API you can just simply create the resources, in a for loop. Add the owner refereces as in samples. And on next reconciliation just read them using `context.getSecondaryResources()` and reconcile them.

https://github.com/java-operator-sdk/java-operator-sdk/blob/affc610de37225d425b8b961f61c2506a2113d30/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Context.java#L18-L18

Using some naming convention in the name of the resource to address / identify them.

**Reply by LarsKrogJensen** [reply]
Oh how nice, was looking at open PR:s but seems to have missed this one.

The Dependent Resource api is really, really nice so I will have a look at the PR and wait for the the 4.1 release.

Thanks a lot
